### PR TITLE
Allow file names to include = when convert to delta

### PR DESCRIPTION
**Pull request by lys0716**
Allow file names to include = when convert to delta

Signed-off-by: Yishuang Lu <luystu@gmail.com>

### CODE REVIEW

**Comment by rahulsmahadev** [comment]
Hi @lys0716 can you rebase your branch with latest delta master.

**Comment by databricks-cla-assistant** [comment]
[CLA assistant check](https://cla.dev.databricks.com/delta-io/delta?pullRequest=264) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.databricks.com/delta-io/delta?pullRequest=264) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: lys0716<br/>:x: Eason Lu<br/><hr/>**Eason Lu** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dev.databricks.com/check/delta-io/delta?pullRequest=264) it.</sub>

**Comment by lys0716** [comment]
@rahulsmahadev Thanks! Updated.

File: src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala [hunk 1]
```diff
@@ -199,8 +199,9 @@ abstract class ConvertToDeltaCommandBase(
     val dateFormatter = DateFormatter()
     val timestampFormatter =
       TimestampFormatter(timestampPartitionPattern, java.util.TimeZone.getDefault)
+    val dir = if (file.isDir) file.getPath else file.getPath.getParent
     val (partitionOpt, _) = PartitionUtils.parsePartition(
-      path,
```
Comment (rahulsmahadev) [inline src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala]: if path is no longer used, we do not need to have it.

Comment (lys0716) [reply src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala]: The path store the absolute path to a file. It is used later to create an AddFile action. But we don't need it to parse the partition in the directory.

Comment (lys0716) [reply src/main/scala/org/apache/spark/sql/delta/commands/ConvertToDeltaCommand.scala]: Hi @rahulsmahadev please let me know if you have other comments on this pr.

**Comment by rahulsmahadev** [comment]
LGTM can one of @liwensun or @brkyvz  have another look as well

**Pull request closed without merging**